example/frontend-testing/rest.js

```diff
@@ -0,0 +1,197 @@
+
+        // Check if the response is successful
+        if (!response.ok) {
+            throw new Error(`Error: ${response.status}`);
+        } else {
+            // Update the response container with the service status
+            responseContainer.textContent = 'Authorized - Service is running';
+        }
+    } catch (error) {
+        // Display error message in case of request failure
+        responseContainer.textContent = 'Could not reach API / API Key missing';
+    }
+}
+
+/**
+ * Save the API key to localStorage
+ * @param {string} apiKey - The API key to save
+ */
+function saveApiKey() {
+    const apiKey = document.getElementById('apiKey').value;
+    if (apiKey) {
+        localStorage.setItem('apiKey', apiKey);
+        const responseContainer = document.getElementById('checkHealth');
+        responseContainer.textContent = "API Key saved successfully!";
+    } else {
+        const responseContainer = document.getElementById('checkHealth');
+        responseContainer.textContent = "Please enter a valid API key first.";
+    }
+}
+
+function allowDrop(event) {
+    event.preventDefault();  // Allow file to be dropped
+}
+
+function handleDrop(event) {
+    event.preventDefault();
+    const files = event.dataTransfer.files;
+    if (files.length > 0) {
+        selectedFile = files[0];
+        document.getElementById('dropZone').textContent = `Selected file: ${selectedFile.name}`;
+    }
+}
+
+function handleFileSelect(event) {
+    const files = event.target.files;
+    if (files.length > 0) {
+        selectedFile = files[0];
+        document.getElementById('dropZone').textContent = `Selected file: ${selectedFile.name}`;
+    }
+}
+
+document.getElementById('dropZone').addEventListener('click', () => {
+    document.getElementById('fileInput').click();  // Open file selector dialog
+});
+
+/**
+ * Function to start transcription by sending a selected file to the API.
+ * Displays the response or error in the 'response' element.
+ */
+async function requestTranscription() {
+    const responseContainer = document.getElementById('response');
+
+    // Check if a file is selected
+    if (!selectedFile) {
+        responseContainer.textContent = 'Please upload a file first.';
+        return;
+    }
+
+    const formData = new FormData();
+
+    const apiKey = localStorage.getItem('apiKey');  // Retrieve the saved API key
+    if (!apiKey) {
+        responseContainer.textContent = 'API key is missing. Please save your API key first.';
+        return;
+    }
+
+    // Verify selected file is valid
+    if (selectedFile === null || selectedFile === undefined) {
+        responseContainer.textContent = 'No file selected.';
+        return;
+    }
+    
+    // Append necessary data to FormData object
+    formData.append('file', selectedFile);
+    formData.append('test', 'test');
+    formData.append('language', 'en');
+    formData.append('settings', '{"test": "test"}');
+    formData.append('model', 'large-v3');
```

Comment:
is this needed or required? This seems like it would make it not usable for lower end machines, no?

Reply:
large-v3 is running by default, might need to revisit